Generic iframe guidance (#88016)
Azure/azure-content: articles/active-directory/develop/reference-third-party-cookies-spas.md

### A note on iframe apps

> A common pattern in web apps is to use an iframe to embed one app inside another. The common assumption from app developers is that, top-level frame handles authenticating the user, and the application hosted in the iframe is expected to trust that the user is signed in and fetch the tokens silently. However, there are couple of caveats to this assumption irrespective of whether third-party cookies are enabled or blocked in the browser.

Reviewer: ```suggestion
A common pattern in web apps is to use an iframe to embed one app inside another. The common assumption from app developers is that the top-level frame handles authenticating the user, and the application hosted in the iframe is expected to trust that the user is signed in and fetch the tokens silently. However, there are couple of caveats to this assumption irrespective of whether third-party cookies are enabled or blocked in the browser.
```

1. Applications running in an iframe (regardless of protocol) will not be able to do to redirect-based interaction, as this is blocked by eSTS for security reasons. 

>     - The one exception is B2C applications using the Embedded Login feature: https://docs.microsoft.com/en-us/azure/active-directory-b2c/embedded-login?pivots=b2c-custom-policy 

Reviewer 2: Let's use the unlocalized relative path here -- should be this

```suggestion
    - The one exception is B2C applications using the [Embedded Login feature](articles/active-directory-b2c/embedded-login.md). 
```

2. Applications running in an iframe must use silent or popup-based interactions for SSO. 

>     - Popup based interaction requires MSAL.js. Our guidance to invoke the authentication from MSAL.js via popup method can be found [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/iframe-usage.md#using-msal-in-iframed-apps). 

Reviewer: ```suggestion
    - Guidance to invoke authentication from MSAL.js via popup method can be found [here](articles/active-directory/develop/scenario-spa-acquire-token.md). 
```

>     - Apart from MSAL.js library, none of the other MSAL flavors(MSAL.Java or MSAL.net etc.) do not provide a way to invoke the authetication via popup method. 

Reviewer: I don't think this line is relevant. The rest of the doc only calls out MSAL.js v2 so we can stay focused on MSAL.js library. 
```suggestion
```

>     - Popups from applications running in iframes may blocked, either due to how they are opened, if the user has popups blocked in their browser, or if the iframe is sandboxed to disallow popups.

Reviewer: ```suggestion
    - Popups have a risk of being blocked due to a user's browser settings or if the iframe is sandboxed by the parent application to disallow popups.
```

>     - Silent login/SSO will fail if third-party cookies are blocked (and the application will need to fallback to popup interaction).

Reviewer: ```suggestion
    - Silent login/SSO can fail for a number of reasons, including if third-party cookies are blocked. Applications will need to fallback to an interactive method. For iframe applications, they should fall back to a popup interaction.
```
